aws-wafv2-loggingconfiguration/src/main/java/software/amazon/wafv2/loggingconfiguration/BaseHandlerStd.java

```diff
@@ -0,0 +1,91 @@
+  @Override
+  public final ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+    final AmazonWebServicesClientProxy proxy,
+    final ResourceHandlerRequest<ResourceModel> request,
+    final CallbackContext callbackContext,
+    final Logger logger) {
+    return handleRequest(
+      proxy,
+      request,
+      callbackContext != null ? callbackContext : new CallbackContext(),
+      proxy.newProxy(ClientBuilder::getClient),
+      logger
+    );
+  }
+
+  protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+    final AmazonWebServicesClientProxy proxy,
+    final ResourceHandlerRequest<ResourceModel> request,
+    final CallbackContext callbackContext,
+    final ProxyClient<Wafv2Client> proxyClient,
+    final Logger logger);
+
+
+  /**
+   * Create Logging Configurations since one doesnt exist already
+   * @param proxy
+   * @param proxyClient
+   * @param model
+   * @param callbackContext
+   * @return Progress Event Object which will proceed to enable delete on termination
+   */
+  protected ProgressEvent<ResourceModel, CallbackContext> putLoggingConfiguration(final AmazonWebServicesClientProxy proxy,
+          final ProxyClient<Wafv2Client> proxyClient,
+          final ResourceModel model,
+          final CallbackContext callbackContext) {
+
+      return proxy.initiate("AWS-WAFv2-LoggingConfiguration::Create::PutLoggingConfiguration", proxyClient, model, callbackContext)
+              .translateToServiceRequest(Translator::translateToCreateRequest)
+              .makeServiceCall((cbRequest, cbProxyClient) -> cbProxyClient.injectCredentialsAndInvokeV2(cbRequest, cbProxyClient.client()::putLoggingConfiguration))
+              .handleError(this::handleError)
+              .progress();
+  }
+
+  public ProgressEvent<ResourceModel, CallbackContext> handleError(final Wafv2Request request,
+          final Exception e,
+          final ProxyClient<Wafv2Client> proxyClient,
+          final ResourceModel resourceModel,
+          final CallbackContext callbackContext) {
+
+      BaseHandlerException ex;
+
+      if (e instanceof WafLimitsExceededException || e instanceof WafServiceLinkedRoleErrorException) {
+          // we can retry this because it's just throttling.
+          throw new CfnThrottlingException(e);
+          // Convert all other exceptions to CFN Exceptions so we can get an error code
+      } else if (e instanceof WafNonexistentItemException ) {
+          ex = new CfnNotFoundException(e);
+      } else if (e instanceof WafInvalidParameterException || e instanceof WafInvalidOperationException ) {
+          ex = new CfnInvalidRequestException(e);
+      } else if (e instanceof WafInternalErrorException) {
+          ex = new CfnInternalFailureException(e);
+      } else {
+          ex = new CfnGeneralServiceException(e);
+      }
+      return ProgressEvent.failed(resourceModel, callbackContext, ex.getErrorCode(), ex.getMessage());
+  }
```

Comment:
We are currently translating the WAF exceptions to error codes here : https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-wafv2/blob/main/aws-wafv2-commons/src/main/java/com/amazonaws/wafv2/commons/ExceptionTranslationWrapper.java

FYI `WafLimitsExceededException` and `WafServiceLinkedRoleErrorException` shouldn't be considered as throttling exceptions (as you can see in the exception translation wrapper).

Reply:
The CFN team has advocated that we rewrap the exceptions that the SDK throws with CFN exceptions. I have used the same Mapping that the Commons package has, but have wrapped it with CFN exceptions.

Reply:
I have tested this - here is a sample error message below - the Request ID is preserved. In addition, a request token is added to the message. So, as discussed, I will keep this the way it is. 

`Resource handler returned message: "1 validation error detected: Value '[errorstring]' at 'loggingConfiguration.redactedFields.1.member.jsonBody.matchPattern.includedPaths' failed to satisfy constraint: Member must satisfy constraint: [Member must have length less than or equal to 512, Member must have length greater than or equal to 1, Member must satisfy regular expression pattern: ([/])|([/](([^~])|(~[01]))+)] (Service: Wafv2, Status Code: 400, Request ID: 6f7de36b-1393-44e3-95b3-bb82be6d90ab, Extended Request ID: null)" (RequestToken: 9242e8dc-8f84-cab5-f0ea-ea6e839a1e99, HandlerErrorCode: GeneralServiceException)`